### PR TITLE
[DOXIA-734] Restore handling of XHTML 1.0 obsolete attributes

### PR DESCRIPTION
**Pull request by kwin**
@michael-o This is what I have in mind, at least for the most important obsolete attributes

### CODE REVIEW

File: doxia-core/src/main/java/org/apache/maven/doxia/parser/Xhtml1BaseParser.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    private static final class AttributeMapping {
+        private final String sourceName;
+        private final String targetName;
+        private final UnaryOperator<String> valueMapper;
+        private final MergeSemantics mergeSemantics;
+
+        enum MergeSemantics {
+            OVERWRITE,
+            IGNORE,
+            PREPEND
```
Comment (michael-o) [inline doxia-core/src/main/java/org/apache/maven/doxia/parser/Xhtml1BaseParser.java:49]: Do we need append as well?

Comment (kwin) [reply doxia-core/src/main/java/org/apache/maven/doxia/parser/Xhtml1BaseParser.java:49]: Let's add once we have a use case

**Review by michael-o — COMMENTED** [review]
Very nice

**Comment by kwin** [comment]
> Does it make sense to log a warning to tell the user that the attribute should be migrated?

I don't think, because the user is complying with the spec while using XHTML5 only attributes/elements (while supported) is not specified for XDoc 2.0. We should rather deprecate XDoc usage alltogether (but this is separate from this issue).

**Pull request merged**